examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -21,6 +21,115 @@
+
+    CachedMessage(const CachedMessage & message) :
+        mPriority(message.mPriority), mMessageControl(message.mMessageControl), mStartTime(message.mStartTime),
+        mDuration(message.mDuration), mMessageText(message.mMessageText), mOptions(message.mOptions)
+    {
+        memcpy(mMessageIdBuffer, message.mMessageIdBuffer, kMessageIdLength);
+        mMessageId = chip::ByteSpan(mMessageIdBuffer, kMessageIdLength);
+
+        for (CachedMessageOption & entry : mOptions)
+        {
+            mResponseOptions.push_back(entry.GetMessageOption());
+        }
+    };
+
+    CachedMessage & operator=(const CachedMessage & message) { return *this; };
+
+    CachedMessage(const chip::ByteSpan & messageId, const chip::app::Clusters::Messages::MessagePriorityEnum & priority,
+                  const chip::BitMask<chip::app::Clusters::Messages::MessageControlBitmap> & messageControl,
+                  const chip::app::DataModel::Nullable<uint32_t> & startTime,
+                  const chip::app::DataModel::Nullable<uint16_t> & duration, std::string messageText) :
+        mPriority(priority),
+        mMessageControl(messageControl), mStartTime(startTime), mDuration(duration), mMessageText(messageText)
+    {
+        if (messageId.size() != kMessageIdLength)
+        {
+            ChipLogProgress(Zcl, "CachedMessage bad message id size, ignoring");
+            return;
+        }
+        memcpy(mMessageIdBuffer, messageId.data(), kMessageIdLength);
+        mMessageId = chip::ByteSpan(mMessageIdBuffer, kMessageIdLength);
+    }
+
+    void AddOption(CachedMessageOption option)
+    {
+        mOptions.push_back(option);
+        mResponseOptions.push_back(option.GetMessageOption());
+    }
+
+    chip::app::Clusters::Messages::Structs::MessageStruct::Type GetMessage()
+    {
+        if (mResponseOptions.size() > 0)
+        {
+            chip::app::DataModel::List<chip::app::Clusters::Messages::MessageResponseOption> options(mResponseOptions.data(),
+                                                                                                     mResponseOptions.size());
+            chip::app::Clusters::Messages::Structs::MessageStruct::Type message{ mMessageId,
+                                                                                 mPriority,
+                                                                                 mMessageControl,
+                                                                                 mStartTime,
+                                                                                 mDuration,
+                                                                                 chip::CharSpan::fromCharString(
+                                                                                     mMessageText.c_str()),
+                                                                                 chip::MakeOptional(options) };
+            return message;
+        }
+        else
```

Comment:
linter would complain (but probably does not run on this file): no else after return.

Reply:
addressed